### DISCUSSION TITLE
Additional Measurement Outputs from OpenFAST via zeromq

### DISCUSSION THREAD

**Discussion by fredrikfselheim** [NatLabRockies/ROSCO#357] [Announcements]
Hi,

I'm currently integrating a fatigue-damage code with OpenFAST for a digital twin framework, using zeromq for process communication. I'm successfully retrieving the rootMOOP (out-of-plane moment) from the wfc_zmq_server class in the control interface. However, I need to extend the measurements to include additional bending moments at critical points:

- Blade roots: RootFzb1 (axial force), RootMxb1 (edgewise), RootMyb1 (flapwise) for each blade
- Tower base: TwrBsFzt (axial force), TwrBsMxt (side-side), TwrBsMyt (fore-aft)

These measurements are crucial for our fatigue analysis. Could you advise if it's possible to access these via the wfc_zmq_server? If not, could you recommend another method to capture these measurements in real-time from OpenFAST?

Thank you for any guidance you can provide!

**Comment by abhineet-gupta** [comment]
Hi,

We currently support the following set of measurements and set-points for the zeromq interface.
[wfc_interface.yaml](https://github.com/NREL/ROSCO/blob/main/rosco/controller/rosco_registry/wfc_interface.yaml).

However, other variables can be added but would require some source code changes.
ROSCO [Local varaibles](https://github.com/NREL/ROSCO/blob/main/rosco/controller/rosco_registry/rosco_types.yaml) are easy to add using [this](https://github.com/NREL/ROSCO/blob/main/rosco/controller/rosco_registry/write_wfc_interface.py) code. 
However, other variables would require extensive source code changes.

You can also look at [this](https://github.com/OpenFAST/openfast/pull/2153) ongoing work which allows real-time access to OpenFAST simulation data.

Abhineet
